config/config.go

```diff
@@ -244,6 +234,28 @@ func (s *SourceSubnet) Validate() error {
 		}
 	}
 
+	// Validate the allowed destinations
+	for _, chainIDs := range s.SupportedDestinations {
+		if _, err := ids.FromString(chainIDs); err != nil {
+			return fmt.Errorf("invalid chainID in source subnet configuration. Provided ID: %s", chainIDs)
+		}
+	}
+
+	// Store the allowed destinations for future use
+	s.supportedDestinations = set.Set[ids.ID]{}
+	for _, blockchainID := range s.SupportedDestinations {
+		chainID, err := ids.FromString(blockchainID)
+		if err != nil {
+			return fmt.Errorf("invalid chainID in configuration. error: %v", err)
+		}
+		if !destinationChainIDs.Contains(blockchainID) {
+			return fmt.Errorf("configured source subnet %s has a supported destination blockchain ID %s that is not configured as a destination blockchain",
+				s.SubnetID,
+				blockchainID)
+		}
+		s.supportedDestinations.Add(chainID)
```

Comment:
I'd consider moving setting the `supportedDestinations` value to a `initialize` method to be more obvious, but no strong preference either way

Reply:
I think that might be more trouble than it's worth. To do so properly, we'd want to iterate through all supported destinations in the validate method, then do so again to cache them in initialize. Here, we do it in a single pass.